### ISSUE TITLE
Confusing error when returning Annotated from a function where the return type is a type var

### ISSUE THREAD

**Issue by FasterSpeeding** [microsoft/pyright#3798]
**Describe the bug**
Pyright complains about a type mismatch when returning a `typing.Annotated` which wraps a type variable from a function typed as returning `type[_T]` despite the two types it reprs visually matching.

**To Reproduce**
* Lint the provided example code using strict pyright targeting Py 3.9.
* Watch it give the following error:

```py
C:\Users\Lucina\PycharmProjects\tanjun\test.py
  C:\Users\Lucina\PycharmProjects\tanjun\test.py:7:12 - error: Expression of type "Type[_T@foo]" cannot be assigned to return type "Type[_T@foo]"
    Type "Type[_T@foo]" cannot be assigned to type "Type[_T@foo]" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
```

**Expected behavior**
Likely that it doesn't complain about this case at all like MyPy.

**Screenshots or Code**
```python
import typing

_T = typing.TypeVar("_T")


def foo(value: type[_T]) -> type[_T]:
    return typing.Annotated[value, "ok"]
```

**VS Code extension or command-line**
Pyright v1.1.265 (CLI tool)
Targeting Python 3.9

**Additional context**
My use case for this is defining some classes where the metaclass' `__getitem__` adds some Annotated metadata to the passed type (also while MyPy does understand this example code, its special casing actually doesn't work for my use case since the metaclass `__getitem__` override is ignored specifically in the context of a type-hint for whatever reason).

**Comment by erictraut** [comment]
You're not using `Annotated` in the correct manner. It is meant to be used in a type expression (annotation), not in a value expression like you are using it here. Please refer to [PEP 593](https://peps.python.org/pep-0593/) for more information about how it is meant to be used.

**Comment by FasterSpeeding** [comment]
> You're not using Annotated in the correct manner. It is meant to be used in a type expression (annotation), not in a value expression like you are using it here.

I'm curious as to where you think a use case like the following would fall under this

```py
class _MarkedMeta(type):
    def __getitem__(self, values: tuple[type[_T], Callable]) -> type[_T]:
        type_ = values[0]
        return Annotated[_T, _Mark(values[1])]


class Marked(metaclass=_MarkedMeta):
    def __new__(cls) -> NoReturn:
        raise RuntimeError("Ranged cannot be initialised")
```

Like the intent here is to form a type hint but python's semantics around what counts as a type vs value seem pretty blurred since at runtime types are just another form of value.

**Comment by erictraut** [comment]
`Annotated` is a special form designed for static analysis use cases. You're treating it as a runtime class here. This is not how it is designed to be used.

I'll also point out that special forms are generally not type compatible with `type`.

Pyright is correct to report an error in this code.

**Comment by FasterSpeeding** [comment]
> I'll also point out that special forms are generally not type compatible with type.
>
> Pyright is correct to report an error in this code.

Could Pyright's errors be improved to indicate what it disagrees with here instead of just saying `Type "Type[_T@foo]" cannot be assigned to type "Type[_T@foo]"` at least (since this is pretty confusing as is)?

> Annotated is a special form designed for static analysis use cases. You're treating it as a runtime class here. This is not how it is designed to be used.

I was of the assumption that it was more for putting runtime metadata in type-hints and tbh I still don't really get this runtime class distinction here when the intended use case is just to use the result of the getitem call as a static type hint (since that's all that'd work with type checkers). (I'm not going to ask anything else, just wanted to try and understand what you're getting at here).
